### DISCUSSION TITLE
TS011F_PLUG_1 No Energy is shown

### DISCUSSION THREAD

**Discussion by conradhagemans** [Koenkk/zigbee2mqtt#19949] [Devices]
I recently added 2 new Tuya Plugs of type TS011F_Plug_1 to my network. Model: _TZ3000_typdpbpg
They do run out of the box perfectly for switching on/off.

I also want to monitor the power usage in Home Assistant.
But I noticed that the Energy value stays on 0 Null. Even after hours of running.

```
{
    "child_lock": "UNLOCK",
    "current": 92,
    "indicator_mode": "off/on",
    "last_seen": "2023-12-01T12:20:31+01:00",
    "linkquality": 18,
    "power": 4,
    "power_outage_memory": "off",
    "state": "ON",
    "voltage": 245,
    "energy": null,
    "update": {
        "installed_version": -1,
        "latest_version": -1,
        "state": null
    },
    "update_available": null
}

```

**Comment by MikZenn** [comment]
Hi, i have this problem after today update.
Do you have the problem also last day o you try add them today after update?

**Reply by conradhagemans** [reply]
I just added today for the first time. So after the update.
I have about 30 devices in my zigbee network. Various power plugs. But this is the first time I am adding the Tuya Plug 1 types.

**Comment by conradhagemans** [comment]
I just checked the logfile. I noticed this error:

```
Zigbee2MQTT:error 2023-12-01 17:24:27: Failed to configure 'Tuya Plug 02', attempt 2 (Error: ConfigureReporting 0xa4c1381ac1e3314f/1 haElectricalMeasurement([{"attribute":"activePower","minimumReportInterval":5,"maximumReportInterval":3600,"reportableChange":10}], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'INVALID_DATA_TYPE')
    at Endpoint.checkStatus (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:421:28)
    at Endpoint.configureReporting (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:762:22)
    at Object.activePower (/app/node_modules/zigbee-herdsman-converters/src/lib/reporting.ts:204:5)
    at Object.configure (/app/node_modules/zigbee-herdsman-converters/src/devices/tuya.ts:3290:17)
    at Configure.configure (/app/lib/extension/configure.ts:117:13))
```

**Comment by CoccoAndrea** [comment]
same problem:

Failed to configure 'Piano Induzione', attempt 2 (Error: ConfigureReporting 0xa4c13846ffc4448e/1 haElectricalMeasurement([{"attribute":"activePower","minimumReportInterval":5,"maximumReportInterval":3600,"reportableChange":10}], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'INVALID_DATA_TYPE') at Endpoint.checkStatus (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:421:28) at Endpoint.configureReporting (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:762:22) at Object.activePower (/app/node_modules/zigbee-herdsman-converters/src/lib/reporting.ts:204:5) at Object.configure (/app/node_modules/zigbee-herdsman-converters/src/devices/tuya.ts:3290:17) at Configure.configure (/app/lib/extension/configure.ts:117:13))

**Comment by Koenkk** [comment]
Issue should be fixed in the latest [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Comment by silbernagel** [comment]
I have a few plugs of the same type.  TS011F / _TZ3000_typdpbpg / TS011F_plug_1.

 I unpaired it from Z2M, stopped the release version, installed edge, and re-paired the plug. Configuration works great in edge (as opposed to the stable branch), but energy is still reported as "null" in Z2M. Any suggestions, or anything I can contribute with for investigation?

**Comment by davidvpe** [comment]
I got two plugs today 

Zigbee Model: TS011F
Zigbee Manufacturer: _TZ3000_okaz9tjs
Manufacturer: [TuYa](https://www.zigbee2mqtt.io/supported-devices/#v=TuYa)
Model: [TS011F_plug_1](https://www.zigbee2mqtt.io/devices/TS011F_plug_1.html#tuya-ts011f_plug_1)


This is the error I am getting

Failed to configure 'smart_plug_oven', attempt 3 (Error: ConfigureReporting 0x6bd204602bd5b370/1 haElectricalMeasurement([{"attribute":"activePower","minimumReportInterval":5,"maximumReportInterval":3600,"reportableChange":10}], {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'INVALID_DATA_TYPE') at Endpoint.checkStatus (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:421:28) at Endpoint.configureReporting (/app/node_modules/zigbee-herdsman/src/controller/model/endpoint.ts:762:22) at Object.activePower (/app/node_modules/zigbee-herdsman-converters/src/lib/reporting.ts:204:5) at Object.configure (/app/node_modules/zigbee-herdsman-converters/src/devices/tuya.ts:3290:17) at Configure.configure (/app/lib/extension/configure.ts:117:13))

Also a Typescript Developer, so if any help is needed I can jump in. But regarding Zigbee I am just starting, so not really sure what to do.

**Reply by Koenkk** [reply]
Could you check with the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)?

**Reply by KibaVega** [reply]
On **Zigbee2MQTT Edge** I have the same problem...

**Reply by davidvpe** [reply]
@Koenkk I just deployed and added the device. Error did not appear this time, Energy shows as N/A kWh so I might need to wait a bit, I will report back in a few hours.

**Reply by davidvpe** [reply]
@Koenkk it’s still showing N/A as Energy, is there anything else i can provide to help?

**Reply by Koenkk** [reply]
could you provide the data/database.db entry of your device?

**Reply by davidvpe** [reply]
```json
{
   "id":2,
   "type":"Router",
   "ieeeAddr":"0x6bd204602bd5b370",
   "nwkAddr":53330,
   "manufId":4660,
   "manufName":"_TZ3000_okaz9tjs",
   "powerSource":"Mains (single phase)",
   "modelId":"TS011F",
   "epList":[
      1
   ],
   "endpoints":{
      "1":{
         "profId":260,
         "epId":1,
         "devId":256,
         "inClusterList":[
            0,
            6,
            3,
            4,
            5,
            4096,
            10,
            57345,
            57344,
            1794,
            2820
         ],
         "outClusterList":[
            
         ],
         "clusters":{
            "genBasic":{
               "attributes":{
                  "hwVersion":0,
                  "manufacturerName":"_TZ3000_okaz9tjs",
                  "zclVersion":3,
                  "appVersion":112,
                  "modelId":"TS011F",
                  "powerSource":1
               }
            },
            "genOnOff":{
               "attributes":{
                  "32768":0,
                  "onOff":1,
                  "tuyaBacklightMode":1,
                  "moesStartUpOnOff":2,
                  "onTime":0,
                  "offWaitTime":0
               }
            },
            "haElectricalMeasurement":{
               "attributes":{
                  "rmsVoltage":231,
                  "rmsCurrent":0,
                  "activePower":0
               }
            },
            "seMetering":{
               "attributes":{
                  "currentSummDelivered":[
                     0,
                     2
                  ]
               }
            }
         },
         "binds":[
            {
               "cluster":6,
               "type":"endpoint",
               "deviceIeeeAddress":"0xbc026efffe2c18f7",
               "endpointID":1
            }
         ],
         "configuredReportings":[
            {
               "cluster":6,
               "attrId":0,
               "minRepIntval":0,
               "maxRepIntval":65000,
               "repChange":1,
               "manufacturerCode":null
            }
         ],
         "meta":{
            
         }
      }
   },
   "appVersion":112,
   "stackVersion":2,
   "hwVersion":0,
   "swBuildId":"0122052017",
   "zclVersion":3,
   "interviewCompleted":true,
   "meta":{
      "configured":-1843040142
   },
   "lastSeen":1703835606576,
   "defaultSendRequestWhen":"immediate"
}
```

**Reply by Koenkk** [reply]
Linking to https://github.com/Koenkk/zigbee-herdsman-converters/pull/6731, do the current/power and voltage measurements work?

**Reply by davidvpe** [reply]
Current, power and voltage seem to be reporting fine 
![IMG_3137](https://github.com/Koenkk/zigbee2mqtt/assets/1934159/e98bdb2e-f714-414f-afe6-113636f39427)

**Reply by Koenkk** [reply]
Could you check again with the latest dev branch?

Changes will be available in the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html) in a few hours from now.

**Reply by DataGhost** [reply]
Just a small FYI this current is not "fine", it's showing 31 amps for 3W heh. But that's due to the bug/regression that should be fixed in that new dev branch a few hours from now, where it'll tell you 0.031 instead.

**Reply by davidvpe** [reply]
It might take me some time, but if the latest-dev docker image is updated I will be able to test right away

**Reply by Koenkk** [reply]
`latest-dev` has just been updated with it

**Reply by davidvpe** [reply]
It’s reporting Energy as 0, all other values look good, current is better also. I guess I need to wait a bit, buts looks better than before 

![image](https://github.com/Koenkk/zigbee2mqtt/assets/1934159/19ba9a62-5f8b-42de-8f89-224f009d2466)

**Reply by davidvpe** [reply]
it took some time but energy seem to be reporting fine now. Thanks! 

![image](https://github.com/Koenkk/zigbee2mqtt/assets/1934159/74d99fa0-0aea-4674-984b-9f10474423d8)

**Reply by KibaVega** [reply]
It's working. Thank You!

![image](https://github.com/Koenkk/zigbee2mqtt/assets/119057206/fcbeb770-bbbc-4a53-82b9-3967574d79c4)

**Comment by silbernagel** [comment]
Here is mine. The seMetering-values seem to be there, and are correct.


``` {
   "id":8,
   "type":"Router",
   "ieeeAddr":"0xa4c1382afef5d97e",
   "nwkAddr":924,
   "manufId":4107,
   "manufName":"_TZ3000_typdpbpg",
   "powerSource":"Mains (single phase)",
   "modelId":"TS011F",
   "epList":[
      1,
      242
   ],
   "endpoints":{
      "1":{
         "profId":260,
         "epId":1,
         "devId":81,
         "inClusterList":[
            0,
            3,
            4,
            5,
            6,
            10,
            57344,
            1794,
            2820,
            4096,
            6280
         ],
         "outClusterList":[
            25
         ],
         "clusters":{
            "genBasic":{
               "attributes":{
                  "hwVersion":0,
                  "manufacturerName":"_TZ3000_typdpbpg",
                  "zclVersion":3,
                  "appVersion":112,
                  "modelId":"TS011F",
                  "powerSource":1
               }
            },
            "genOnOff":{
               "attributes":{
                  "32768":0,
                  "onOff":1,
                  "tuyaBacklightMode":1,
                  "moesStartUpOnOff":0,
                  "onTime":0,
                  "offWaitTime":0
               }
            },
            "haElectricalMeasurement":{
               "attributes":{
                  "rmsVoltage":228,
                  "rmsCurrent":4388,
                  "activePower":970
               }
            },
            "seMetering":{
               "attributes":{
                  "currentSummDelivered":[
                     0,
                     10426
                  ]
               }
            }
         },
         "binds":[
            {
               "cluster":6,
               "type":"endpoint",
               "deviceIeeeAddress":"0xe0798dfffe9f2298",
               "endpointID":1
            },
            {
               "cluster":2820,
               "type":"endpoint",
               "deviceIeeeAddress":"0xe0798dfffe9f2298",
               "endpointID":1
            }
         ],
         "configuredReportings":[
            {
               "cluster":6,
               "attrId":0,
               "minRepIntval":0,
               "maxRepIntval":65000,
               "repChange":1,
               "manufacturerCode":null
            }
         ],
         "meta":{
            
         }
      },
      "242":{
         "profId":41440,
         "epId":242,
         "devId":97,
         "inClusterList":[
            
         ],
         "outClusterList":[
            33
         ],
         "clusters":{
            
         },
         "binds":[
            
         ],
         "configuredReportings":[
            
         ],
         "meta":{
            
         }
      }
   },
   "appVersion":112,
   "stackVersion":2,
   "hwVersion":0,
   "swBuildId":"1.0.5\u0000",
   "zclVersion":3,
   "interviewCompleted":true,
   "meta":{
      "configured":-1843040142
   },
   "lastSeen":1703841383177,
   "defaultSendRequestWhen":"immediate"
}
```

**Comment by smart-ctrl** [comment]
@Koenkk I have eight _TZ3000_typdpbpg plugs, energy is reported when detected as TS011F_Plug_1

![image](https://github.com/Koenkk/zigbee2mqtt/assets/85099988/a5ca444f-13af-47e0-9bc2-e1456cb1513c)

**Comment by silbernagel** [comment]
Updated to latest edge now, and it reports correctly! Much thanks! Sending you a beer or two!

**Comment by smart-ctrl** [comment]
@Koenkk It's not possible to calibrate Current, no change in Current value?
![image](https://github.com/Koenkk/zigbee2mqtt/assets/85099988/8ebfe2df-7b85-4ed5-b258-67faaa7f0e00)

**Comment by smart-ctrl** [comment]
@Koenkk I've noticed a problem, 3 of my _TZ3000_typdpbpg plugs reporting Current correct. But another 3 plugs show half of the correct Current value. See debug log below, the right Current value should be 0.117. They show the right Current value in ZHA, but the other 3 plugs show the wrong Current value instead. So wich one is faulty and how to provide a fix for it? It's not possible to use Current calibration, as I meantion earlier it's not working at the moment.

bug 2024-01-01 17:03:38Received Zigbee message from '0xa*305', type 'attributeReport', cluster 'haElectricalMeasurement', data '{"rmsCurrent":117}' from endpoint 1 with groupID 0
Info 2024-01-01 17:03:38MQTT publish: topic 'zigbee2mqtt/0xa*305', payload '{"child_lock":"UNLOCK","current":0.0585,"energy":0,"indicator_mode":"off/on","linkquality":255,"power":21,"power_outage_memory":"off","state":"ON","update":{"installed_version":-1,"latest_version":-1,"state":null},"update_available":null,"voltage":223}'

**Reply by Koenkk** [reply]
Can you provide the data/database.db entry of the wrong ones?

**Reply by smart-ctrl** [reply]
@Koenkk Yes..

{"id":23,"type":"Router","ieeeAddr":"0xa*","nwkAddr":46232,"manufId":4107,"manufName":"_TZ3000_typdpbpg","powerSource":"Mains (single phase)","modelId":"TS011F","epList":[1,242],"endpoints":{"1":{"profId":260,"epId":1,"devId":81,"inClusterList":[0,3,4,5,6,10,57344,1794,2820,4096,6280],"outClusterList":[25],"clusters":{"genBasic":{"attributes":{"hwVersion":0,"manufacturerName":"_TZ3000_typdpbpg","zclVersion":3,"appVersion":112,"modelId":"TS011F","powerSource":1}},"genOnOff":{"attributes":{"32768":0,"onOff":0,"tuyaBacklightMode":1,"moesStartUpOnOff":0,"onTime":0,"offWaitTime":0}},"haElectricalMeasurement":{"attributes":{"rmsVoltage":224,"rmsCurrent":117,"activePower":0,"acCurrentDivisor":2000,"acCurrentMultiplier":1}},"seMetering":{"attributes":{"currentSummDelivered":[0,11],"divisor":100,"multiplier":1}}},"binds":[{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0xe*","endpointID":1},{"cluster":2820,"type":"endpoint","deviceIeeeAddress":"0xe*","endpointID":1},{"cluster":1794,"type":"endpoint","deviceIeeeAddress":"0xe*","endpointID":1}],"configuredReportings":[{"cluster":6,"attrId":0,"minRepIntval":0,"maxRepIntval":65000,"repChange":1,"manufacturerCode":null},{"cluster":2820,"attrId":1285,"minRepIntval":5,"maxRepIntval":3600,"repChange":5,"manufacturerCode":null},{"cluster":2820,"attrId":1288,"minRepIntval":5,"maxRepIntval":3600,"repChange":50,"manufacturerCode":null},{"cluster":1794,"attrId":0,"minRepIntval":5,"maxRepIntval":3600,"repChange":[1,1],"manufacturerCode":null}],"meta":{}},"242":{"profId":41440,"epId":242,"devId":97,"inClusterList":[],"outClusterList":[33],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":112,"stackVersion":2,"hwVersion":0,"swBuildId":"0122052017","zclVersion":3,"interviewCompleted":true,"meta":{"configured":1310285083},"lastSeen":1704132866927,"defaultSendRequestWhen":"immediate"}

**Reply by Koenkk** [reply]
What is the data/database.db entry of the plugs reporting energy correctly?

**Reply by smart-ctrl** [reply]
@Koenkk

{"id":26,"type":"Router","ieeeAddr":"0xa*","nwkAddr":42959,"manufId":4107,"manufName":"_TZ3000_typdpbpg","powerSource":"Mains (single phase)","modelId":"TS011F","epList":[1,242],"endpoints":{"1":{"profId":260,"epId":1,"devId":81,"inClusterList":[0,3,4,5,6,10,57344,1794,2820,4096,6280],"outClusterList":[25],"clusters":{"genBasic":{"attributes":{"modelId":"TS011F","manufacturerName":"_TZ3000_typdpbpg","powerSource":1,"zclVersion":3,"appVersion":112,"stackVersion":2,"hwVersion":0,"swBuildId":"0122052017"}},"haElectricalMeasurement":{"attributes":{"acCurrentDivisor":2000,"acCurrentMultiplier":1,"rmsVoltage":226,"rmsCurrent":222,"activePower":24}},"seMetering":{"attributes":{"divisor":100,"multiplier":1,"currentSummDelivered":[0,0]}},"genOnOff":{"attributes":{"32768":0,"onOff":1,"tuyaBacklightMode":1,"moesStartUpOnOff":0,"onTime":0,"offWaitTime":0}}},"binds":[{"cluster":6,"type":"endpoint","deviceIeeeAddress":"0x*","endpointID":1},{"cluster":2820,"type":"endpoint","deviceIeeeAddress":"0x*","endpointID":1},{"cluster":1794,"type":"endpoint","deviceIeeeAddress":"0x*","endpointID":1}],"configuredReportings":[{"cluster":2820,"attrId":1285,"minRepIntval":5,"maxRepIntval":3600,"repChange":5,"manufacturerCode":null},{"cluster":2820,"attrId":1288,"minRepIntval":5,"maxRepIntval":3600,"repChange":50,"manufacturerCode":null},{"cluster":1794,"attrId":0,"minRepIntval":5,"maxRepIntval":3600,"repChange":[1,1],"manufacturerCode":null}],"meta":{}},"242":{"profId":41440,"epId":242,"devId":97,"inClusterList":[],"outClusterList":[33],"clusters":{},"binds":[],"configuredReportings":[],"meta":{}}},"appVersion":112,"stackVersion":2,"hwVersion":0,"swBuildId":"0122052017","zclVersion":3,"interviewCompleted":true,"meta":{"configured":1310285083},"lastSeen":1704387674387,"defaultSendRequestWhen":"immediate"}

**Reply by Koenkk** [reply]
For `"id":23,` (the one with wrong current), what shows up when you read the swBuildId:

<img width="955" alt="Screenshot 2024-01-04 at 19 40 42" src="https://github.com/Koenkk/zigbee2mqtt/assets/2892853/d89d7698-8415-4402-a8bc-163b61a66294">

**Reply by smart-ctrl** [reply]
@Koenkk
![image](https://github.com/Koenkk/zigbee2mqtt/assets/85099988/2a62472d-b31e-4d1d-ad40-865c8b76e354)

**Reply by Koenkk** [reply]
Every attribute is the same so we cannot distinguish it, that means we can only use the `current_calibration`, it should be set to `50` in this case. Z2M 1.35.0 contains some fixes for this, can you try with that? Note that after setting you need to wait until the next report comes in.

**Reply by smart-ctrl** [reply]
@Koenkk No, Current calibration still not working in 1.35.0 with this plug, doesn't change the Current value.

**Reply by Koenkk** [reply]
Found the issue and pushed a fix, setting `current_calibration` to `-50` should do it.

Changes will be available in the [dev branch](https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html) in a few hours from now.

**Reply by smart-ctrl** [reply]
@Koenkk -50 doesn't work, but 100 works fine. Thanks for the fix.
![image](https://github.com/Koenkk/zigbee2mqtt/assets/85099988/e1ec7a01-6e88-4f98-afb3-ff3a2ce09394)

**Reply by Koenkk** [reply]
It needs to be doubled of course, great! 😄

**Comment by Gerben321** [comment]
Are you guys able to actually use these plugs? Mine keeps turning itself off randomly. Everything seems to report just fine though. Update shows -1 like in the comments above, so no option to update via OTA.

**Reply by conradhagemans** [reply]
Sure. I am using these working perfect:
https://a.aliexpress.com/_EGAsc9f

**Reply by Gerben321** [reply]
Which version is that exactly reported in z2m? Mine looks identical. Reported as _TZ3000_okaz9tjs (TS011F_plug_1). But it randomly turns of, like other people have also reported. Can't find updates for the firmware.

**Reply by conradhagemans** [reply]
Mine is: 
_TZ3000_typdpbpg
